app/routes/index.jsx

```diff
@@ -76,13 +78,33 @@ export default function Index() {
   const fetcher = useFetcher();
   const [shouldFetch, setShouldFetch] = useState(true);
   const [page, setPage] = useState(2);
+  const [isLoading, setIsLoadign] = useState(true);
   const [searchParams, ] = useSearchParams();
+  const location = useLocation();
+  const navigate = useNavigate();
 
   const onFetchMore = () => {
     if(!shouldFetch) return;
     fetcher.load(`/?index&${searchParams.toString()}&page=${page}`);
   };
 
+  useEffect(() => {
+    if(location.hash.includes('questionId')){
+      const hashQuestion = location.hash;
+      const questionId = hashQuestion.substring(hashQuestion.indexOf('questionId=') + 11, hashQuestion.length);
+      if(parseInt(questionId,10)){
+        navigate(`/questions/${questionId}`);
+      }
+      else {
+        setIsLoadign(false);
+        //To-Do Redirect to Not Found Question Page.
```

Comment:
Think we will redirect to a 404 with #85 through the loader.

Reply:
I'll delete comment